manticore/native/memory.py

```diff
@@ -1165,16 +1179,30 @@ def write(self, address, value, force=False):
                     self._symbols.setdefault(base + offset, []).append((condition, value[offset]))
         else:
 
+            # regroups into areas to minimize calls to super().write
+            areas = []
+            startArea = 0
+            sizeArea = 0
             for offset in range(size):
                 if issymbolic(value[offset]):
+                    if sizeArea > 0:
+                        areas.append((startArea, sizeArea))
+                        sizeArea = 0
+                        startArea = offset + 1
                     if not self.access_ok(address + offset, "w", force):
                         raise InvalidMemoryAccess(address + offset, "w")
                     self._symbols[address + offset] = [(True, value[offset])]
                 else:
-                    # overwrite all previous items
-                    if address + offset in self._symbols:
-                        del self._symbols[address + offset]
```

Comment:
I think that the CI fails with a timeout because we removed this. It might be that it ends up in an infinite loop.

Also, the `test_mix_of_concrete_and_symbolic` test fails.

In this test we:
1. Create memory with 0, 2, 4, 6 offsets/indexes/memory cells as concrete values and 1, 3, 5, 7 as symbolic.
2. Assert if those concretes/symbolic addresses are really concrete/symbolic.
3. Swap indexes of concrete/symbolic memory cells and reinitialise memory bytes to concrete/symbolic values.
4. We reassert those. **Here, the test fails.**

This is because we don't delete the old symbolic mappings, so in `mem._symbols` we have old 1, 3, 5, 7 symbolic cells and 0, 2, 4, 6 new symbolic cells. **The 1, 3, 5, 7 should not be there.**

Reply:
I am going to try fixing that by readding the `del ...` and see if it works.

Reply:
Indeed